src/pages/signUp/signUpSteps/Welcome.tsx

```diff
@@ -10,39 +7,27 @@ import {
   FacebookLogin,
   GoogleLogin,
   LinkedInLogin,
+  PasswordInputField,
   TextInputField,
 } from "components";
 
-
 const SignUp: FunctionComponent = () => {
-  const theme = useTheme();
   const { isValid, values } = useFormikContext();
-  const { newPassword, confirmPassword } = values as FormikValues; 
-  const [showPassword, setShowPassword] = useState(false);
-  const [showPasswordIcon, setShowPasswordIcon] = useState(false);
-
-  useAuthenticatedRedirect();
+  const { newPassword, confirmPassword } = values as FormikValues;
+  const [showEndAdornment, setShowEndAdornment] = useState(false);
+  const [maskPassword, setMaskPassword] = useState(true);
 
-  useEffect(()=> {
-    newPassword || confirmPassword ? setShowPasswordIcon(true) : setShowPasswordIcon(false);
-  },[newPassword, confirmPassword]);
-
-  const togglePassword = () => {
-    setShowPassword(!showPassword);
+  const togglePasswordMask = () => {
+    setMaskPassword(!maskPassword);
   };
 
-  const PasswordEndAdornment = () => {
-    const PasswordIcon = showPassword ? VisibilityOffIcon : VisibilityIcon;
+  useAuthenticatedRedirect();
 
-    return showPasswordIcon ? (
-      <IconButton
-        aria-label="Change password field visible status"
-        onClick={ togglePassword }
-      >
-        <PasswordIcon sx={ { color: theme.colors.white } } />
-      </IconButton>
-    ) : null;
-  };
+  useEffect(() => {
+    newPassword || confirmPassword
+      ? setShowEndAdornment(true)
+      : setShowEndAdornment(false);
+  }, [newPassword, confirmPassword]);
```

Comment:
Would it be more straightforwad to just use 

```
const showEndAdornment = !!(newPasword || confirmPassword)
```
instead of using `useState` and `useEffect`?

Reply:
Well i usually like to complicate things so no haha. Good catch, i was trying to do other logic inside the useEffect that is no longer applicable and forgot to clean this up